scRNA-seq-advanced/setup/integration-setup.Rmd

```diff
@@ -0,0 +1,566 @@
+
+combined_sce
+```
+
+
+
+## Pre-integration visualization
+
+To get a sense of our data _before_ integrating, let's look at UMAPs for each individual library and the combined SCE.
+
+
+```{r}
+plot_library_umap <- function(sce, sce_name) {
+  scater::plotReducedDim(sce,
+                         dimred = "UMAP", 
+                         point_size = 0.5, 
+                         point_alpha = 0.5) +
+    ggplot2::ggtitle(sce_name)
+}
+
+umap_list <- purrr::imap(
+  sce_list,
+  plot_library_umap
+)
+
+cowplot::plot_grid(plotlist = umap_list, nrow = 2)
+
+# In the same dimesion space:
+scater::plotReducedDim(combined_sce,
+                       dimred = "UMAP",
+                       colour_by = "sample",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.2) + 
+  ggplot2::ggtitle("UMAPs separately calculated")
+
+
+scater::plotReducedDim(combined_sce,
+                       dimred = "combined_UMAP",
+                       colour_by = "sample",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.2) +
+  ggplot2::ggtitle("UMAPs calculated on combined data")
+  
+
+```
+
+
+## Integration
+
+To start, we'll use `fastMNN` from the `batchelor` package:
+
+```{r}
+integrated_sce <- batchelor::fastMNN(combined_sce,
+                                     # Specify the groups:
+                                     batch = colData(combined_sce)$sample,
+                                     # Optionally specify genes to use as the previously-ID'd 
+                                     # high variance genes
+                                     subset.row = hvg_list
+                                     )
+
+integrated_sce
+```
+- The `reconstructed` assay are corrected expression values. 
+Make sure to not use them for DE though!
+
+- The `corrected` reduced dimensions are the PCs resulting from the corrected expression values. 
+We can calculate UMAP embeddings from the PCs.
+
+
+```{r}
+# Specify the object's `reducedDim` name
+integrated_sce <- scater::runUMAP(integrated_sce, dimred = "corrected")
+integrated_sce
+```
+
+We can also/alternatively add this information into our `combined_sce` object to keep everything in one place:
+
+
+```{r}
+reducedDim(combined_sce, "fastmnn_PCA") <- reducedDim(integrated_sce, "corrected")
+reducedDim(combined_sce, "fastmnn_UMAP") <- reducedDim(integrated_sce, "UMAP")
+```
+
+
+#### The before/after reveal
+
+```{r}
+# Before
+scater::plotReducedDim(combined_sce,
+                       dimred = "combined_UMAP",
+                       colour_by = "sample",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.4) +
+  ggplot2::ggtitle("Before integration")
+
+
+# After
+scater::plotReducedDim(combined_sce,
```

Comment:
_Yes._ I actually had started writing a second section about "boy this is kind of annoying how points are on top of each other!" and showing how you could shuffle and plot again, but I thought it might be overkill here. Thinking about it again now maybe it's a useful trick to show either as a non-live chunk, or in exercises.